lib/ecto/repo/preloader.ex

```diff
@@ -168,18 +177,15 @@ defmodule Ecto.Repo.Preloader do
     query = update_in query.select.expr, &{:{}, [], [field, &1]}
```

Comment:
So, ecto doesn't have any problem with this but when using my gist if I happen to come across a polymorphic relationship where two records have the same `id` but were fetched in different tables, I believe one will overwrite the other and that may be a very subtle bug to discover. Any ideas on how that might work?